From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: chickeneer <emcchickeneer@gmail.com>
Date: Sun, 3 Nov 2024 10:14:19 -0600
Subject: [PATCH] ArrowKnockbackEvent

An event for altering the knockback data value
post weapon enchants allowing other context to alter it

diff --git a/src/main/java/com/empireminecraft/customevents/ArrowKnockbackEvent.java b/src/main/java/com/empireminecraft/customevents/ArrowKnockbackEvent.java
new file mode 100644
index 0000000000000000000000000000000000000000..f24f220cb850ee474d66b143933a0e38a0f85a82
--- /dev/null
+++ b/src/main/java/com/empireminecraft/customevents/ArrowKnockbackEvent.java
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2024. Starlis LLC / dba Empire Minecraft
+ *
+ * This source code is proprietary software and must not be redistributed without Starlis LLC's approval
+ */
+
+package com.empireminecraft.customevents;
+
+import org.bukkit.entity.Entity;
+import org.bukkit.event.HandlerList;
+import org.bukkit.event.entity.EntityEvent;
+import org.bukkit.inventory.ItemStack;
+import org.jetbrains.annotations.NotNull;
+
+public class ArrowKnockbackEvent extends EntityEvent {
+    private static final HandlerList handlers = new HandlerList();
+    private final Entity target;
+    private double knockback;
+
+    public ArrowKnockbackEvent(@NotNull Entity arrow, @NotNull Entity target, double knockback) {
+        super(arrow);
+        this.target = target;
+        this.knockback = knockback;
+    }
+
+    @NotNull
+    public Entity getTarget() {
+        return this.target;
+    }
+
+    public void setKnockback(double knockback) {
+        this.knockback = knockback;
+    }
+
+    public double getKnockback() {
+        return knockback;
+    }
+
+    @NotNull
+    public HandlerList getHandlers() {
+        return handlers;
+    }
+
+    @NotNull
+    public static HandlerList getHandlerList() {
+        return handlers;
+    }
+}
